### ISSUE TITLE
constructor with enum parameter crashes

### ISSUE THREAD

**Issue by archaeron** [mbdavid/LiteDB#1860]
**Version**
5.0.9

**Describe the bug**
I have a class with a single, public constructor that takes an enum as a parameter.
The code crashes when it tries to get that class from the Database.
The same thing works without a public constructor by making all fields nullable.

**Code to Reproduce**
```csharp
#nullable enable

using System;
using System.Linq;
using LiteDB;

namespace litedb_example
{
    [Serializable]
    public enum EnumAB
    {
        [System.Runtime.Serialization.EnumMember(Value = @"a")]
        A,
        [System.Runtime.Serialization.EnumMember(Value = @"b")]
        B,
    }

    public class C1
    {
        public int Id { get; set; }

        public EnumAB? EnumAB {get; set; }
    }

    public class C2
    {
        public int Id { get; set; }

        public EnumAB EnumAB {get; set; }

        public C2(int id, EnumAB enumAB)
        {
            Id = id;
            EnumAB = enumAB;
        }
    }

    class Program
    {
        static void Main(string[] args)
        {

            // Open database (or create if doesn't exist)
            using (var db = new LiteDatabase(@"./example.db"))
            {
                // Get a collection (or create, if doesn't exist)
                var col1 = db.GetCollection<C1>("c1");
                var col2 = db.GetCollection<C2>("c2");

                var c1 = new C1
                {
                    Id = 1,
                    EnumAB = EnumAB.A
                };

                col1.Insert(c1);

                var c2 = new C2
                (
                    id: 1,
                    enumAB: EnumAB.A
                );

                col2.Insert(c2);

                var value1 = col1.FindAll().FirstOrDefault();
                System.Console.WriteLine(value1);

                var value2 = col2.FindAll().FirstOrDefault();
                System.Console.WriteLine(value2);
            }
        }
    }
}
```

**Expected behavior**
I would expect either both variants to fail or both to work.

**Screenshots/Stacktrace**
Unhandled exception. LiteDB.LiteException: Failed to create instance for type 'litedb_example.C2' from assembly 'litedb_example.C2, litedb-example, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. Checks if the class has a public constructor with no parameters.
 ---> System.ArgumentException: Type 'litedb_example.C2' does not have a default constructor (Parameter 'type')
   at System.Linq.Expressions.Expression.New(Type type)
   at LiteDB.Reflection.CreateClass(Type type)
   at LiteDB.Reflection.CreateInstance(Type type)
   --- End of inner exception stack trace ---
   at LiteDB.Reflection.CreateInstance(Type type)
   at LiteDB.BsonMapper.<>c__DisplayClass58_0.<Deserialize>b__0(BsonDocument v)
   at LiteDB.BsonMapper.Deserialize(Type type, BsonValue value)
   at LiteDB.LiteQueryable`1.<ToEnumerable>b__27_2(BsonDocument x)
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Boolean& found)
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source)

**Comment by archaeron** [comment]
adding `[BsonCtor]` to the constructor of `C2` has the same effect

**Comment by for7raid** [comment]
The exception says `Checks if the class has a public constructor with no parameters.`
Please take a look at docs - https://github.com/mbdavid/LiteDB/wiki/Object-Mapping#mapper-conventions - the class must have parameterless constructor.
So this issue does not relate to enum type. Make an empty constructor in C2 class.

**Comment by archaeron** [comment]
@for7raid you're looking at the docs for v4 (it says so at the top of the document you linked)

This is not true anymore for v5, see here:
https://www.litedb.org/docs/object-mapping/

**Comment by archaeron** [comment]
this should be fixed by #1867 
thank you @for7raid !

**Comment by qsgy** [comment]
[Serializable]
    public enum EnumAB
    {
        [System.Runtime.Serialization.EnumMember(Value = @"a")]
        A,
        [System.Runtime.Serialization.EnumMember(Value = @"b")]
        B,
    }
这个不起作用, 序列化后的数据还是A和B,不是a和b
